ui/page/root/overview_page.go

```diff
@@ -912,20 +929,76 @@ func (pg *OverviewPage) txStakingSection(gtx C) D {
-						return pg.pageContentWrapper(gtx, "Recent Transactions", nil, func(gtx C) D {
-							return pg.centerLayout(gtx, values.MarginPadding10, values.MarginPadding10, func(gtx C) D {
-								gtx.Constraints.Min.X = gtx.Constraints.Max.X
-								return pg.Theme.Body1("No recent transaction").Layout(gtx)
+						return pg.pageContentWrapper(gtx, values.String(values.StrRecentTransactions), nil, func(gtx C) D {
+							if len(pg.transactions) == 0 {
+								return pg.centerLayout(gtx, values.MarginPadding10, values.MarginPadding10, func(gtx C) D {
+									gtx.Constraints.Min.X = gtx.Constraints.Max.X
+									return pg.Theme.Body1(values.String(values.StrNoTransactions)).Layout(gtx)
+								})
+							}
+
+							return pg.recentTransactions.Layout(gtx, len(pg.transactions), func(gtx C, index int) D {
+								row := components.TransactionRow{
+									Transaction: pg.transactions[index],
+									Index:       index,
+								}
+
+								return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										return components.LayoutTransactionRow(gtx, pg.Load, row, false)
+									}),
+									layout.Rigid(func(gtx C) D {
+										// No divider for last row
+										if row.Index == len(pg.transactions)-1 {
+											return layout.Dimensions{}
+										}
+
+										gtx.Constraints.Min.X = gtx.Constraints.Max.X
+										separator := pg.Theme.Separator()
+										return layout.E.Layout(gtx, func(gtx C) D {
+											// Show bottom divider for all rows except last
+											return layout.Inset{Left: values.MarginPadding8}.Layout(gtx, separator.Layout)
+										})
+									}),
+								)
 							})
 						})
 					})
 				}),
 				layout.Flexed(.5, func(gtx C) D {
-					return pg.pageContentWrapper(gtx, "Staking Activity", nil, func(gtx C) D {
-						return pg.centerLayout(gtx, values.MarginPadding10, values.MarginPadding10, func(gtx C) D {
-							gtx.Constraints.Min.X = gtx.Constraints.Max.X
+					return pg.pageContentWrapper(gtx, values.StrStakingActivity, nil, func(gtx C) D {
+						if len(pg.stakes) == 0 {
+							return pg.centerLayout(gtx, values.MarginPadding10, values.MarginPadding10, func(gtx C) D {
+								gtx.Constraints.Min.X = gtx.Constraints.Max.X
+
+								return pg.Theme.Body1(values.StrNoStaking).Layout(gtx)
+							})
+						}
+
+						return pg.recentStakes.Layout(gtx, len(pg.stakes), func(gtx C, index int) D {
+							row := components.TransactionRow{
+								Transaction: pg.stakes[index],
+								Index:       index,
+							}
 
-							return pg.Theme.Body1("No recent Staking Activity").Layout(gtx)
+							return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+								layout.Rigid(func(gtx C) D {
+									return components.LayoutTransactionRow(gtx, pg.Load, row, false)
+								}),
+								layout.Rigid(func(gtx C) D {
+									// No divider for last row
+									if row.Index == len(pg.stakes)-1 {
+										return layout.Dimensions{}
```

Comment:
```suggestion
										return D{}
```

We can agree to use D{} instead of layout.Dimensions{}